packages/components/src/internal/components/picklist/actions.ts

```diff
@@ -0,0 +1,233 @@
+                    {fieldKey: "SampleID/Created"},
+                    {fieldKey: "SampleID/CreatedBy"},
+                    {fieldKey: "SampleID/StorageLocation"},
+                    {fieldKey: "SampleID/StorageRow"},
+                    {fieldKey: "SampleID/StorageCol"},
+                    {fieldKey: "SampleID/isAliquot"},
+                ]
+            }]
+        };
+        return Ajax.request({
+            url: buildURL('query', 'saveQueryViews.api'),
+            method: 'POST',
+            jsonData,
+            success: Utils.getCallbackWrapper(response => {
+                resolve(response);
+            }),
+            failure: Utils.getCallbackWrapper(response => {
+                console.error(response);
+                reject("There was a problem creating the default view for the picklist. " + resolveErrorMessage(response));
+            }),
+        });
+    });
+}
+
+export function createPicklist(name: string, description: string, shared: boolean) : Promise<CreatePicklistResponse> {
+    return new Promise((resolve, reject) => {
+        Domain.create({
+            domainDesign: {
+                name: name,
+                fields: [
+                    {
+                        name: 'SampleID',
+                        rangeURI: 'int',
+                        required: true,
+                        lookupContainer: LABKEY.container.path,
+                        lookupSchema: SCHEMAS.INVENTORY.SAMPLE_ITEMS.schemaName,
+                        lookupQuery: SCHEMAS.INVENTORY.SAMPLE_ITEMS.queryName,
+                    },
+                ],
+                indices: [
+                    {
+                        columnNames: ['SampleID'],
+                        unique: true,
+                    }
+                ]
+            },
+            kind: PICKLIST,
+            options: {
+                keyName: 'id',
+                keyType: 'AutoIncrementInteger',
+                description,
+                category: shared ? PUBLIC_PICKLIST_CATEGORY : PRIVATE_PICKLIST_CATEGORY
+            },
+            success: (response) => { resolve({name: response.name, domainId: response.domainId}); },
+            failure: (err) => { reject(err); }
+        });
+    });
+}
+
+export function updatePicklist(picklist: PicklistModel) :Promise<PicklistModel>  {
+    return new Promise((resolve, reject) => {
+        fetchListDesign(picklist.listId)
+            .then(listDesign => {
+                let domain = listDesign.domain;
+                let options = {
+                    domainId: domain.domainId,
+                    keyName: 'id',
+                    keyType: 'AutoIncrementInteger',
+                    description : picklist.Description,
+                    category: picklist.Category,
+                };
+                if (picklist.name !== domain.get('name')) {
+                    options = Object.assign(options, {name: picklist.name});
+                }
```

Comment:
Would this be the same as the following?
```suggestion
                const options = {
                    ...listDesign.options,
                    name: picklist.name
                    description : picklist.Description,
                    category: picklist.Category,
                };
```

Also, is there any reason that the `name prop of picklist is lowercase but `Category` and `Description` are capitalized?

Reply:
There doesn't seem to be an `options` field on `listDesign`, but I can get rid of the `if` statement.

The difference in capitalization is simply because that was the casing returned by the server and I got lazy tracking down why that was. Case-sensitivity is fun.